apis/llms_resource.py

```
@@ -14,29 +15,30 @@ def data_generator(response):
             # pass in data to completion function, unpack data
             response = completion(**data)
             if data['stream'] == True: 
                 return Response(data_generator(response), mimetype='text/event-stream')
-            return response, 200 
-        except openai.error.OpenAIError as e:
+        except OpenAIError as e:
             # Handle OpenAI API errors
             error_message = str(e)
-            print(f"OpenAI API Error: {error_message}")
-            return jsonify({"error": error_message}), 500
+            app.logger.error(f"OpenAI API Error: {error_message}")
```

Ruff (0.13.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)